### PR TITLE
vim-patch:d2ea7cf10a4d

### PR DESCRIPTION
**Pull request by clason**
Update runtime files
https://github.com/vim/vim/commit/d2ea7cf10a4d026ebd402594d656af7d5c811c24

omit `runtime/doc/if_tcl.txt`
omit `runtime/doc/textprop.txt`
omit `runtime/tutor/*`
omit `runtime/syntax/vim.vim` (cherry-picked in https://github.com/neovim/neovim/commit/2dd7828511d04a8b7f1ac4331c719a751a5db869)

manual merge of `runtime/pack/dist/opt/termdebug/plugin/termdebug.vim`

### CODE REVIEW

**Comment by clason** [comment]
As the person guilty of cherry-picking runtime updates, I'll be trying to catch up with the vim runtime updates; then we should consider how we could organize following upstream's two-week rhythm more closely.

(lint failure is from `screen.c` being whacked again with the `uncrustify` stick so unrelated)

**Comment by clason** [comment]
CI fails on `oldtest`, specifically this line https://github.com/neovim/neovim/blob/dce50312e1e9af81fb0e3b61d6e70bdf286fbffb/src/nvim/testdir/test_syntax.vim#L125
which no longer matches. Changing that line to
```vim
  call assert_match(' \d*\.\d* \+[^0]\d* .* cppNumber*', a)
```
makes that test pass, but I don't understand how it passes without that change for upstream. Any ideas? @jamessan

**Comment by clason** [comment]
(asan test failure looks due to flakiness.)

File: src/nvim/testdir/test_syntax.vim [hunk 1]
```diff
@@ -116,13 +116,13 @@ func Test_syntime()
...
 
-  view ../memfile_test.c
+  view samples/memfile_test.c
   setfiletype cpp
   redraw
   let a = execute('syntime report')
   call assert_match('^  TOTAL *COUNT *MATCH *SLOWEST *AVERAGE *NAME *PATTERN', a)
   call assert_match(' \d*\.\d* \+[^0]\d* .* cppRawString ', a)
-  call assert_match(' \d*\.\d* \+[^0]\d* .* cppNumber ', a)
+  call assert_match(' \d*\.\d* \+[^0]\d* .* cppNumber', a)
```
Comment (gpanders) [inline src/nvim/testdir/test_syntax.vim:125]: The trailing space should remain I think.

**Comment by clason** [comment]
Ok, CI looks as good to me as it is going to get. Feel free to merge unless I forgot something (in which case feel free to tell me off), then I'll tackle the next one.

**Pull request merged**